src/i2c.rs

```diff
@@ -550,6 +550,10 @@ impl PinScl<FMPI2C> for PF15<AlternateOD<AF4>> {}
 pub enum Error {
     OVERRUN,
     NACK,
+    TIMEOUT,
+    BUS,
+    CRC,
+    ARBITRATION,
```

Comment:
Hm, I was about to move all errors over to `embedded-error`. But I can also do that independently.